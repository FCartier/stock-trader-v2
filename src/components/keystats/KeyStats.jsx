import React from "react";
import { SectionHeader } from "../reusables/SectionHeader";

const KeyStats = props => {
  const { keystats } = props;

  return Object.keys(keystats).length ? (
    <div>
        <div className="keystats">
          <SectionHeader title="KEY STATS" />
          <div className="keystats-row">
            <div className="keystats-column">
              <ul className="keystats-list">
                <li>
                  <p>Previous Close</p> <h5>{keystats.close}</h5>
                </li>
                <li>
                  <p>Day Range</p>{" "}
                  <h5>
                    {keystats.low}-{keystats.high}
                  </h5>
                </li>
                <li>
                  <p>Volume</p> <h5>{keystats.latestVolume}</h5>
                </li>
              </ul>
            </div>
            <div className="keystats-column">
              <ul className="keystats-list">
                <li>
                  <p>Open </p> <h5>{keystats.open}</h5>
                </li>
                <li>
                  <p>52 Week Range</p>{" "}
                  <h5>
                    {keystats.week52Low}-{keystats.week52High}
                  </h5>
                </li>
                <li>
                  <p> Total Avg. Volume</p> <h5>{keystats.avgTotalVolume}</h5>
                </li>
              </ul>
            </div>
          </div>
        </div>
    </div>
  ) : null
};

export default KeyStats;
