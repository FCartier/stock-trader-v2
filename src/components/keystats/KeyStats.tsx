import * as React from "react";
import { SectionHeader } from "../reusables/SectionHeader";
import { Row, Column, List } from "./styles";
import { fetchStatus } from "../reusables/fetchStatus";
import NoData from "../reusables/NoData";

interface IKeyStatsProps {
  keystats?: {
    close: number;
    low: number;
    high: number;
    open: number;
    latestVolume: number;
    week52High: number;
    week52Low: number;
    avgTotalVolume: number;
  };
  status?: fetchStatus;
}

const KeyStats: React.SFC<IKeyStatsProps> = ({ keystats, status }) => {
  return status !== fetchStatus.pending ? (
    <div>
      <SectionHeader title="KEY STATS" />
      {status === fetchStatus.success ? (
        <Row>
          <Column>
            <List>
              <li>
                <p>Previous Close</p> <h5>{keystats.close}</h5>
              </li>
              <li>
                <p>Day Range</p>{" "}
                <h5>
                  {keystats.low}-{keystats.high}
                </h5>
              </li>
              <li>
                <p>Volume</p> <h5>{keystats.latestVolume}</h5>
              </li>
            </List>
          </Column>
          <Column>
            <List>
              <li>
                <p>Open </p> <h5>{keystats.open}</h5>
              </li>
              <li>
                <p>52 Week Range</p>{" "}
                <h5>
                  {keystats.week52Low}-{keystats.week52High}
                </h5>
              </li>
              <li>
                <p> Total Avg. Volume</p> <h5>{keystats.avgTotalVolume}</h5>
              </li>
            </List>
          </Column>
        </Row>
      ) : (
        <NoData componentName="key stats" />
      )}
    </div>
  ) : null;
};

export default KeyStats;
