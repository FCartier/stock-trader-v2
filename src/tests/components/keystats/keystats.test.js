import React from 'react';
import KeyStats from '../../../components/keystats/KeyStats';
import renderer from 'react-test-renderer';
import testData from './keystats.data.json'


describe("keystats component", () => {
    test('keystats component should display all data fields', () => {
        const component = renderer.create(
          <KeyStats keystats={testData}/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
})
