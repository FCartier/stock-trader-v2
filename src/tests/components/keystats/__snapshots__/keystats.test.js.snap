// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keystats component keystats component should display all data fields 1`] = `
<div>
  <div
    className="keystats"
  >
    <div
      className="category-header"
    >
      <h3>
        KEY STATS
      </h3>
    </div>
    <div
      className="keystats-row"
    >
      <div
        className="keystats-column"
      >
        <ul
          className="keystats-list"
        >
          <li>
            <p>
              Previous Close
            </p>
             
            <h5>
              51.806
            </h5>
          </li>
          <li>
            <p>
              Day Range
            </p>
             
            <h5>
              51.56
              -
              53.17
            </h5>
          </li>
          <li>
            <p>
              Volume
            </p>
             
            <h5>
              77
            </h5>
          </li>
        </ul>
      </div>
      <div
        className="keystats-column"
      >
        <ul
          className="keystats-list"
        >
          <li>
            <p>
              Open 
            </p>
             
            <h5>
              51.76
            </h5>
          </li>
          <li>
            <p>
              52 Week Range
            </p>
             
            <h5>
              47.31
              -
              64
            </h5>
          </li>
          <li>
            <p>
               Total Avg. Volume
            </p>
             
            <h5>
              32193
            </h5>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`keystats component keystats should display all data fields 1`] = `
<div>
  <div
    className="keystats"
  >
    <div
      className="category-header"
    >
      <h3>
        KEY STATS
      </h3>
    </div>
    <div
      className="keystats-row"
    >
      <div
        className="keystats-column"
      >
        <ul
          className="keystats-list"
        >
          <li>
            <p>
              Previous Close
            </p>
             
            <h5>
              51.806
            </h5>
          </li>
          <li>
            <p>
              Day Range
            </p>
             
            <h5>
              51.56
              -
              53.17
            </h5>
          </li>
          <li>
            <p>
              Volume
            </p>
             
            <h5>
              77
            </h5>
          </li>
        </ul>
      </div>
      <div
        className="keystats-column"
      >
        <ul
          className="keystats-list"
        >
          <li>
            <p>
              Open 
            </p>
             
            <h5>
              51.76
            </h5>
          </li>
          <li>
            <p>
              52 Week Range
            </p>
             
            <h5>
              47.31
              -
              64
            </h5>
          </li>
          <li>
            <p>
               Total Avg. Volume
            </p>
             
            <h5>
              32193
            </h5>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`keystats should display all data fields 1`] = `
<div>
  <div
    className="keystats"
  >
    <div
      className="category-header"
    >
      <h3>
        KEY STATS
      </h3>
    </div>
    <div
      className="keystats-row"
    >
      <div
        className="keystats-column"
      >
        <ul
          className="keystats-list"
        >
          <li>
            <p>
              Previous Close
            </p>
             
            <h5>
              51.806
            </h5>
          </li>
          <li>
            <p>
              Day Range
            </p>
             
            <h5>
              51.56
              -
              53.17
            </h5>
          </li>
          <li>
            <p>
              Volume
            </p>
             
            <h5>
              77
            </h5>
          </li>
        </ul>
      </div>
      <div
        className="keystats-column"
      >
        <ul
          className="keystats-list"
        >
          <li>
            <p>
              Open 
            </p>
             
            <h5>
              51.76
            </h5>
          </li>
          <li>
            <p>
              52 Week Range
            </p>
             
            <h5>
              47.31
              -
              64
            </h5>
          </li>
          <li>
            <p>
               Total Avg. Volume
            </p>
             
            <h5>
              32193
            </h5>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
